R/derive_param_lasta.R

```diff
@@ -0,0 +1,358 @@
+#'   The columns specified in `subject_keys` are expected.
+#'
+#'    *Permitted Values:* a `data.frame()` object
+#'
+#'    *Required or Optional:* Required
+#'
+#' @param filter_source Filter to be applied to `dataset` to derive the
+#'                      Last Disease Assessment
+#'
+#'    *Required or Optional:* Required
+#'
+#' @param order Sort order, after which the last record shall be taken by
+#'              the `by_vars` to determine Last Disease Assessment. Created
+#'              using `admiral::vars()`.
+#'
+#'    *Permitted Values:* an `admiral::vars` object
+#'
+#'    *Default:* `admiral::vars(STUDYID, USUBJID, ADT)`
+#'
+#'    *Required or Optional:* Required
+#'
+#' @param by_vars Grouping columns, the last of which (ordered by `order`)
+#'                shall be taken as the Last Disease Assessment record. Created
+#'                using `admiral::vars()`.
+#'
+#'    *Permitted Values:* an `admiral::vars` object
+#'
+#'    *Default:* `admiral::vars(STUDYID, USUBJID)`
+#'
+#'    *Required or Optional:* Required
+#'
+#' @param source_pd Date of first progressive disease (PD)
+#'
+#'   If the parameter is specified, the observations of the input `dataset` for
+#'   deriving the new parameter are restricted to observations up to the
+#'   specified date. Observations at the specified date are included. For
+#'   subjects without first PD date all observations are take into account.
+#'
+#'   *Permitted Values:* a `date_source` object (see `admiral::date_source()`
+#'   for details)
+#'
+#'   *Default:* `NULL`
+#'
+#'   *Required or Optional:* Optional
+#'
+#' @param source_datasets Source dataframe to be used to calculate the
+#'                        first PD date
+#'
+#'   A named list of dataframes is expected (although for BoR) only one dataframe is
+#'   needed. It links the `dataset_name` from `source_pd` with an existing dataframe.
+#'
+#'   For example if `source_pd = pd_date` with
+#'   ```
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataframe in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#'    *Required or Optional:* Optional
+#'
+#' @param subject_keys Columns to uniquely identify a subject
+#'
+#'   A list of symbols created using `admiral::vars()`.
+#'
+#'   *Default:* `admiral::vars(STUDYID, USUBJID)`
+#'
+#'   *Required or Optional:* Required
+#'
+#' @param set_values_to Columns to set
+#'
+#'   A named list returned by `vars()` defining the columns to be set for the
+#'   new parameter, e.g. `vars(PARAMCD = "LSTAC", PARAM = "Last Disease
+#'    Assessment Censored at First PD by Investigator")` is expected. The values
+#'    must be symbols, character strings, numeric values, or `NA`.
+#'
+#'    *Required or Optional:* Required
+#'
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @examples
+#'
+#' \dontrun{
```

Comment:
You know, I always have my examples as donotrun, just my defauly.  As I am generally accessing external Objects, APIs, AWS Buckets, etc. and they would error otherwise.
I shall remove this tag as it is not needed here.
Good spot.